x/ccv/provider/keeper/keeper.go

```diff
@@ -73,6 +73,28 @@ func NewKeeper(
 	}
 }
 
+// ExposeAllFields exposes (only by value) all the un-exported fields of the ccv provider keeper,
+// purely for testing purposes. DO NOT Use this method outside of testing, as it breaks encapsulation.
+func (k Keeper) ExposeAllFields() (
```

Comment:
Although I see the point you are making, maybe consider making a validator func instead of exposing private fields. And run that validator func inside the provider keeper init procedure - the node should not start if the keepers are not available.

```golang

func (k Keeper) ValidateInit() {
	// make checks on the keeper
	panic("EvidenceKeeprs is not set")
}
```